### PR TITLE
tlclient: init at 4.16.0

### LINKED ISSUE #105399: ThinLinc
**Project description**
ThinLinc is a remote desktop application, like TeamViewer but for Linux. My college uses it to give students remote access to their school linux machines.

There is a generic Linux tarball and a MacOS image downloadable from their website (I have provided links below). The Linux tarball looks like something that would be moved to the filesystem root in a regular distro, and there is a global configuration file (./etc/tlclient.conf) which could be used for configuration options. I would personally find options for "FULL_SCREEN_MODE" and "OPTIONS_POPUP_KEY" particulary useful.

If someone knows how this could be done but doesn't have the time to do it themselves, please leave an explanation or a link to a relevant webpage. I would be happy to become a maintainer of this package, if I just knew the how-tos.

Here is the link to the AUR PKGBUILD for reference: https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=tlclient

If there are any questions, I'll be happy to provide more information

**Metadata**

* homepage URL: https://www.cendio.com/
* source URL: https://www.cendio.com/downloads/clients/tl-4.12.0-6517-client-linux-dynamic-x86_64.tar.gz (Linux) / https://www.cendio.com/downloads/clients/tl-4.12.0_6517-client-macos.iso (MacOS)
* license: nonfree (I think)
* platforms: unix, linux, darwin, (armv7?)

### PR DESCRIPTION
**Pull request by Noah1989**
Based on https://github.com/NixOS/nixpkgs/pull/106052 (no idea why that never got merged), fixes https://github.com/NixOS/nixpkgs/issues/105399

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Rider128** [comment]
I run the client and I have encountered 0 issue, all the functionalities were there.

**Comment by Vaisakhkm2625** [comment]
😥 does this merged soon?

File: pkgs/applications/networking/remote/tlclient/default.nix
Comment (eclairevoyant) [inline pkgs/applications/networking/remote/tlclient/default.nix:1]: Please run `nixfmt` on this file.

Also, this should go in pkgs/by-name...

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -5433,6 +5433,8 @@ with pkgs;
 
   fx = callPackage ../development/tools/fx { };
 
+  tlclient = callPackage ../applications/networking/remote/tlclient { };
```
Comment (eclairevoyant) [inline pkgs/top-level/all-packages.nix:5436]: ...and remove this entry.

File: pkgs/applications/networking/remote/tlclient/default.nix [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      name = "tlclient";
+      exec = "tlclient";
+      icon = "tlclient";
+      comment = meta.description;
+      type = "Application";
+      categories = [ "Network" "RemoteAccess" ];
+    })
+  ];
+
+  meta = with lib; {
```
Comment (eclairevoyant) [inline pkgs/applications/networking/remote/tlclient/default.nix:61]:
```suggestion
  meta = {
```
tracking issue: #292468

File: pkgs/applications/networking/remote/tlclient/default.nix [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    description = "Linux remote desktop client built on open source technology";
+    license = {
+      fullName = "Cendio End User License Agreement 3.2";
+      url = "https://www.cendio.com/thinlinc/docs/legal/eula";
+      free = false;
+    };
+    homepage = "https://www.cendio.com/";
+    maintainers = with maintainers; [ shamilton ];
+    platforms = platforms.linux;
```
Comment (eclairevoyant) [inline pkgs/applications/networking/remote/tlclient/default.nix:70]:
```suggestion
    maintainers = with lib.maintainers; [ shamilton ];
    platforms = [ "x86_64-linux" ];
```

File: pkgs/applications/networking/remote/tlclient/default.nix [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+  src = fetchurl {
+    # x86_64 arch tarball
+    url = "https://www.cendio.com/downloads/clients/tl-${version}-client-linux-dynamic-x86_64.tar.gz";
+    sha256 = "ea3a92186cbaf71cc4ab692b3baeadd94b3836242e58705a38fe3683f4f18691";
+  };
+
+  nativeBuildInputs = [ autoPatchelfHook copyDesktopItems ];
+  propagatedBuildInputs = [ alsa-lib libX11 pcsclite ];
+
```
Comment (eclairevoyant) [inline pkgs/applications/networking/remote/tlclient/default.nix:30]:
```suggestion

  dontConfigure = true;
```

File: pkgs/applications/networking/remote/tlclient/default.nix [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+  version = "4.16.0-3389";
+
+  src = fetchurl {
+    # x86_64 arch tarball
+    url = "https://www.cendio.com/downloads/clients/tl-${version}-client-linux-dynamic-x86_64.tar.gz";
+    sha256 = "ea3a92186cbaf71cc4ab692b3baeadd94b3836242e58705a38fe3683f4f18691";
+  };
+
+  nativeBuildInputs = [ autoPatchelfHook copyDesktopItems ];
+  propagatedBuildInputs = [ alsa-lib libX11 pcsclite ];
```
Comment (eclairevoyant) [inline pkgs/applications/networking/remote/tlclient/default.nix:29]: why are these propagated?

File: pkgs/applications/networking/remote/tlclient/default.nix [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+  };
+in
+stdenv.mkDerivation rec {
+  pname = "tlclient";
+  version = "4.16.0-3389";
+
+  src = fetchurl {
+    # x86_64 arch tarball
+    url = "https://www.cendio.com/downloads/clients/tl-${version}-client-linux-dynamic-x86_64.tar.gz";
+    sha256 = "ea3a92186cbaf71cc4ab692b3baeadd94b3836242e58705a38fe3683f4f18691";
```
Comment (eclairevoyant) [inline pkgs/applications/networking/remote/tlclient/default.nix:25]: please use SRI hash (set `hash = "";` and then get the hash from the error)

**Review by eclairevoyant — CHANGES_REQUESTED** [review]

File: pkgs/applications/networking/remote/tlclient/default.nix (hunk 3 above)
Comment (eclairevoyant) [reply pkgs/applications/networking/remote/tlclient/default.nix:70]: also did you check with @SCOTT-HAMILTON if they are still prepared to maintain this?

**Pull request closed without merging**